### PR TITLE
Add full loader for download business summary

### PR DESCRIPTION
**Pull request by BrandonSharratt**
*Issue:*https://github.com/bcgov/entity/issues/24518

*Description of changes:*
Added full screen modal and made it display when downloading business summary and hide when finished. Note that while there does seem to be a downloading flag for the common template, it doesn't appear to be passed through properly so I can't simply set it to true to cause them all to download and there is quite the chain that I'd have to add it to props and pass it through on so I went this way as it's easier

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namex license (Apache 2.0).

### CODE REVIEW

**Comment by bcregistry-sre** [comment]
Temporary Url for review: https://business-dashboard-dev--pr-94-ziv1i9va.web.app

File: src/components/bcros/businessDetails/Links.vue [hunk 1]
```diff
@@ -243,7 +262,7 @@ const contacts = getContactInfo('registries')
     </BcrosDialog>
 
     <!-- Staff Comments -->
-    <div v-if="hasRoleStaff && currentBusiness && !isDisableNonBenCorps()">
```
Comment (severinbeauvais) [inline src/components/bcros/businessDetails/Links.vue:246]: Don't we want to disable staff comments for non-Ben corps (that should be managed in COLIN instead)?

**Comment by severinbeauvais** [comment]
While testing this, I found a new bug. Try this business (which is actually in Test only). Compare with Filings UI.

https://business-dashboard-dev--pr-94-ziv1i9va.web.app/BC1151476

File: src/components/bcros/businessDetails/Links.vue (hunk 1 above)
Comment (severinbeauvais) [reply src/components/bcros/businessDetails/Links.vue:246]: It's not enabled in Filings UI. LMK if this is a change in requirements.

image

Comment (BrandonSharratt) [reply src/components/bcros/businessDetails/Links.vue]: Accidently chucked this commit when adding a new one, but I didn't change that line

**Pull request closed without merging**